platform/darwin/src/MGLLoggingConfiguration.m

```diff
@@ -0,0 +1,66 @@
+#import "MGLLoggingConfiguration_Private.h"
+
+#if __has_builtin(__builtin_os_log_format)
+#import <os/log.h>
+#endif
+
+@implementation MGLLoggingConfiguration
+
++ (instancetype)sharedConfiguration
+{
+    static dispatch_once_t once;
+    static id sharedConfiguration;
+    dispatch_once(&once, ^{
+        sharedConfiguration = [[self alloc] init];
+        ((MGLLoggingConfiguration *)sharedConfiguration).handler = nil;
+    });
+    return sharedConfiguration;
+}
+
+- (void)setHandler:(void (^)(MGLLoggingLevel, NSString *, NSUInteger, NSString *))handler {
+    
+    if (!handler) {
+        _handler = [self defaultBlockHandler];
+    } else {
+        _handler = handler;
+    }
+}
+
+- (void)logCallingFunction:(const char *)callingFunction functionLine:(NSUInteger)functionLine messageType:(MGLLoggingLevel)type format:(id)messageFormat, ...
+{
+    va_list formatList;
+    va_start(formatList, messageFormat);
+    NSString *formattedMessage = [[NSString alloc] initWithFormat:messageFormat arguments:formatList];
+    va_end(formatList);
+    
+    _handler(type, [NSString stringWithFormat:@"%s", callingFunction], functionLine, formattedMessage);
+    
+}
+
+- (MGLLoggingBlockHandler)defaultBlockHandler {
+    MGLLoggingBlockHandler mapboxHandler = ^(MGLLoggingLevel level, NSString *fileName, NSUInteger line, NSString *message) {
```

Comment:
I think it's fine for fileName to be a `const char*` here.

Reply:
In our last meeting we discussed that in order to simplify 3rd party integration and swift integration we would keep this as `NSString`.

Reply:
For reference: @fabian-guerra and I also discussed using `@ __FILE__` instead, but unfortunately that gives us the full path.

I'd love to avoid two formats here, but I'm not sure we can.